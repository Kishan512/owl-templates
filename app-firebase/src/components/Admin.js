import { Component } from "../../lib/owl";

export class Admin extends Component {
  template = "Admin";
}
