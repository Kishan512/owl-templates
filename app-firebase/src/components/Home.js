import { Component } from "../../lib/owl";

export class Home extends Component {
  template = "Home";
}
