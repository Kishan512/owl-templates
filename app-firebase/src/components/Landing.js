import { Component } from "../../lib/owl";

export class Landing extends Component {
  template = "Landing";
}
