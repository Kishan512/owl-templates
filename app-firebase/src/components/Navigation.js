import { Component, router } from "../../lib/owl";
import { SignOutButton } from "./SignOut";

export class Navigation extends Component {
  template = "Navigation";
  components = { Link: router.Link, SignOutButton };
}
