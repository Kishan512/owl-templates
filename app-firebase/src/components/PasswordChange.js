import { Component } from "../../lib/owl";

const INITIAL_STATE = {
  passwordOne: "",
  passwordTwo: "",
  error: null
};

export class PasswordChange extends Component {
  template = "PasswordChange";

  constructor(parent, props) {
    super(parent, props);
    this.state = Object.assign({}, INITIAL_STATE);
  }

  async onSubmit(event) {
    event.preventDefault();
    try {
      const state = this.state;
      await this.env.firebase.updatePassword(state.passwordOne);
      this.state = Object.assign({}, INITIAL_STATE);
    } catch (error) {
      this.state.error = error;
    }
  }

  get isInvalid() {
    const state = this.state;
    return state.passwordOne !== state.passwordTwo || state.passwordTwo === "";
  }
}
