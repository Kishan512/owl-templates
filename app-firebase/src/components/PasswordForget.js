import { Component, router } from "../../lib/owl";

const INITIAL_STATE = {
  email: "",
  error: null
};

export class PasswordForget extends Component {
  template = "PasswordForget";
  state = Object.assign({}, INITIAL_STATE);

  async onSubmit(event) {
    event.preventDefault();
    try {
      const state = this.state;
      await this.env.firebase.resetPassword(state.email);
      this.state = Object.assign({}, INITIAL_STATE);
    } catch (error) {
      this.state.error = error;
    }
  }
}

export class PasswordForgetLink extends Component {
  template = "PasswordForgetLink";
  components = { Link: router.Link };
}
