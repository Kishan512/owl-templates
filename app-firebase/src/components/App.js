import { Component } from "../../lib/owl";
import { Navigation } from "./Navigation";

export class App extends Component {
  template = "App";
  components = { Navigation };
}
