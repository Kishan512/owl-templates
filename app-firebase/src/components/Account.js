import { PasswordChange } from "./PasswordChange";
import { Component } from "../../lib/owl";

export class Account extends Component {
  template = "Account";
  components = { PasswordChange };
}
