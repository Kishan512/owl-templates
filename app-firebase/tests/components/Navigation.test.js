import { Navigation } from "../../src/components/Navigation";
import { makeTestEnv, makeTestFixture } from "../helpers";

let fixture;
let env;

beforeEach(async () => {
  fixture = makeTestFixture();
  env = await makeTestEnv();
});

afterEach(() => {
  fixture.remove();
  env.cleanup();
});

describe("Navigation", () => {
  test("normal rendering with no user", async () => {
    const navigation = new Navigation(env);
    await navigation.mount(fixture);
    expect(fixture.innerHTML).toMatchSnapshot();
  });

  test("normal rendering with logged user", async () => {
    env.session.authUser = true;
    const navigation = new Navigation(env);
    await navigation.mount(fixture);
    expect(fixture.innerHTML).toMatchSnapshot();
  });
});
