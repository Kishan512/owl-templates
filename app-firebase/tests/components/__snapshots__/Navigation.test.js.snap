// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation normal rendering with logged user 1`] = `"<div><ul><li><a href=\\"/\\" class=\\"router-link-active\\">Landing</a></li><li><a href=\\"/home\\">Home</a></li><li><a href=\\"/account\\">Account</a></li><li><a href=\\"/admin\\">Admin</a></li><li><button> Sign Out </button></li></ul></div>"`;

exports[`Navigation normal rendering with no user 1`] = `"<div><ul><li><a href=\\"/\\" class=\\"router-link-active\\">Landing</a></li><li><a href=\\"/signin\\">Sign In</a></li></ul></div>"`;
